### PR TITLE
Show progressbar during TimeDrive

### PR DESCRIPTION
**Pull request by lang-m**
This PR adds a simple progressbar for TimeDrive runs for `verbose=2`. Progress is estimated base on the total number of steps to save (`n`) and the number of files already written to disk. Estimating progress based on this is not super accurate as the simulation can get slower/faster but it is a useful hint for interactive inside a notebook.

During the simulation:
grafik

After the simulation is finished:
grafik

### CODE REVIEW

File: oommfc/drivers/driver.py [hunk 1]
```diff
@@ -114,9 +116,11 @@ def drive(
...
         verbose : int, optional
 
-            If ``verbose=0``, no output is printed. For ``verbose>=1``
-            information about the OOMMF runner and the runtime is printed to
-            stdout. Defaults is ``verbose=1``.
+            If ``verbose=0``, no output is printed. For ``verbose=1`` information about
+            the OOMMF runner and the runtime is printed to stdout. For ``verbose=2`` a
+            progress bar is displayed for time drives. Note, that this information only
+            relies on the number of magnetisation snapshot already saved to disk and
+            therefore only gives a rough hint. Defaults is ``verbose=1``.
```
Comment (fangohr) [inline oommfc/drivers/driver.py:123]:
```suggestion 
            therefore only gives a rough indication of progress. The default 
            is ``verbose=1``.
```

File: oommfc/oommf/oommf.py [hunk 2]
```diff
@@ -45,9 +50,20 @@ def call(self, argstr, need_stderr=False, n_threads=None, verbose=1):
...
         verbose : int, optional
 
-            If ``verbose=0``, no output is printed. For ``verbose>=1``
-            information about the OOMMF runner and the runtime is printed to
-            stdout. Defaults is ``verbose=1``.
+            If ``verbose=0``, no output is printed. For ``verbose=1`` information about
+            the OOMMF runner and the runtime is printed to stdout. For ``verbose=2`` a
+            progress bar is displayed for time drives. Note, that this information only
+            relies on the number of magnetisation snapshot already saved to disk and
+            therefore only gives a rough hint. Defaults is ``verbose=1``.
```
Comment (fangohr) [inline oommfc/oommf/oommf.py:57]:
```suggestion
            therefore only gives an indication of progress. The default 
            is ``verbose=1``.
```

**Comment by fangohr** [comment]
Very cool idea. Minor changes suggested. 

This PR introduces somewhat significant changes: we start using asyncio, concurrent.futures and [tqdm](https://pypi.org/project/tqdm/) as additional dependency, introducing associated complexity. 

I think it is worth the change - the new tools are standard and tqdm is widely used. And I image the progress bar is psychologically really nice to have.

**Comment by marijanbeg** [comment]
This is very cool!

**Comment by lang-m** [comment]
Closing in favour of #100

**Pull request closed without merging**